src/components/Preferences/LocalIdentities.js

```diff
@@ -22,88 +23,129 @@ import Import from './Import'
-  }
+const LocalIdentities = React.memo(
+  ({ dao, localIdentities, onClearAll, onImport, onModify, onModifyEvent }) => {
+    // transform localIdentities from object into array and attach address to each entry
+    const identities = Object.entries(localIdentities).map(
+      ([address, identity]) => ({
+        ...identity,
+        address,
+      })
+    )
 
-  const { identityEvents$ } = React.useContext(IdentityContext)
-  const { showLocalIdentityModal } = React.useContext(LocalIdentityModalContext)
-  const updateLabel = address => async () => {
-    try {
-      await showLocalIdentityModal(address)
-      // preferences get all
-      onModifyEvent()
-      // for iframe apps
-      identityEvents$.next({ type: identityEventTypes.MODIFY, address })
-    } catch (e) {
-      /* nothing was updated */
+    if (!identities.length) {
+      return <EmptyLocalIdentities onImport={onImport} />
     }
-  }
-  const downloadHref = window.URL.createObjectURL(
-    new Blob([JSON.stringify(identities)], { type: 'text/json' })
-  )
-  // standard: https://en.wikipedia.org/wiki/ISO_8601
-  const today = format(Date.now(), 'yyyy-MM-dd')
-
-  return (
-    <React.Fragment>
-      <Headers>
-        <div>Custom label</div>
-        <div>Address</div>
-      </Headers>
-      <List>
-        {identities.map(({ address, name }) => (
-          <Item key={address}>
-            <Label>{name}</Label>
-            <div>
-              <IdentityBadge
-                entity={address}
-                popoverAction={{
-                  label: 'Edit custom label',
-                  onClick: updateLabel(address),
-                }}
-                popoverTitle={<LocalIdentityPopoverTitle label={name} />}
+
+    const { identityEvents$ } = React.useContext(IdentityContext)
+    const { showLocalIdentityModal } = React.useContext(
+      LocalIdentityModalContext
+    )
+    const updateLabel = address => async () => {
+      try {
+        await showLocalIdentityModal(address)
+        // preferences get all
+        onModifyEvent()
+        // for iframe apps
+        identityEvents$.next({ type: identityEventTypes.MODIFY, address })
+      } catch (e) {
+        /* nothing was updated */
+      }
+    }
+    // standard: https://en.wikipedia.org/wiki/ISO_8601
+    const today = format(Date.now(), 'yyyy-MM-dd')
+    const [addressSelected, setAddressSelected] = React.useState(
+      new Map(identities.map(({ address }) => [address, true]))
+    )
+    const allSelected = Array.from(addressSelected.values()).every(v => v)
+    const someSelected = Array.from(addressSelected.values()).some(v => v)
+    const handleToggleAllSelectedChange = () => {
+      setAddressSelected(
+        new Map(
+          identities.map(({ address }) => [
+            address,
+            !(allSelected || someSelected),
+          ])
+        )
+      )
+    }
+    const handleToggleAddressSelected = address => () =>
```

Comment:
Should we use `useCallback()` for `handleToggleAllSelectedChange()` and `handleToggleAddressSelected()`?